src/main/java/org/zalando/pazuzu/sort/TopologicalSortLinear.java

```diff
@@ -0,0 +1,129 @@
+package org.zalando.pazuzu.sort;
+
+import java.util.*;
+import java.util.function.Function;
+
+/**
+ * Created by dimangaliyev on 25/04/16.
+ * Topological sort algorithm
+ * Running time complexity O(n + m)
+ * where n - number of vertices, m - number of edges
+ */
+public class TopologicalSortLinear<T> {
+
+    /**
+     * Color enum for vertex
+     * WHITE - not visited
+     * GREY - visited in process of traversal
+     * BLACK - visited
+     */
+    private enum Color {
+        WHITE, GREY, BLACK
```

Comment:
Color is misleading, name the enum for what it is used: VisitingState or sth like this.